src/layer/AbstractLayer.ts

```diff
@@ -23,14 +29,15 @@ export class AbstractLayer {
     this.description = description;
   }
 
-  public async getMap(params: GetMapParams, api: ApiType): Promise<Blob> {
+  public async getMap(params: GetMapParams, api: ApiType, reqConfig?: RequestConfiguration): Promise<Blob> {
     switch (api) {
       case ApiType.WMS:
         const url = this.getMapUrl(params, api);
         const requestConfig: AxiosRequestConfig = {
           // 'blob' responseType does not work with Node.js:
           responseType: typeof window !== 'undefined' && window.Blob ? 'blob' : 'arraybuffer',
           useCache: true,
+          ...reqConfig,
```

Comment:
This is an elegant solution, the only problem is that users might find this a shortcut so they can provide the parameters directly to the axios. Not sure how to best solve this though... Specifying the parameters one by one also doesn't sound nice. Do you have some other idea? If not, fftr.

Reply:
The idea is that because RequestConfiguration has a type, users cannot shorcut parametrs of axios request into RequestConfiguration, since this type only allows for certain key-value pairs, such as useCache, retries and cancelToken (if other parameters are specified an exception should be thrown?)
Maybe this doesn't prevent the problem the way I thought so some modifications are needed.

Reply:
Yes, but this only works if developers are using TypeScript, JavaScript will allow anything to be passed. But as I said, I don't see another clean solution, so fftr.